component-tools/src/main/java/org/talend/sdk/component/tools/ComponentValidator.java

```diff
@@ -153,6 +92,59 @@ public void run() {
         components.forEach(c -> log.debug("Found component: " + c));
 
         final Set<String> errors = new LinkedHashSet<>();
+        final Validators.ValidatorHelper helper = new ValidatorHelper() {
+
+            @Override
+            public boolean isService(final Parameter parameter) {
+                return ComponentValidator.this.parameterModelService
+                        .isService(new ParameterModelService.Param(parameter));
+            }
+
+            @Override
+            public ResourceBundle findResourceBundle(final Class<?> component) {
+                return ComponentValidator.this.findResourceBundle(component);
+            }
+
+            @Override
+            public String findPrefix(final Class<?> component) {
+                return ComponentValidator.this
+                        .components(component)
+                        .map(c -> findFamily(c, component) + "." + c.name())
+                        .orElseThrow(() -> new IllegalStateException(component.getName()));
+            }
+
+            @Override
+            public String validateFamilyI18nKey(final Class<?> clazz, final String... keys) {
+                return ComponentValidator.this.validateFamilyI18nKey(clazz, keys);
+            }
+
+            @Override
+            public List<ParameterMeta> buildOrGetParameters(final Class<?> c) {
+                return ComponentValidator.this.buildOrGetParameters(c);
+            }
+
+            @Override
+            public String validateIcon(final Icon annotation, final Collection<String> errors) {
+                return ComponentValidator.this.validateIcon(annotation, errors);
+            }
+
+            @Override
+            public ParameterModelService getParameterModelService() {
+                return ComponentValidator.this.parameterModelService;
+            }
+
+            @Override
+            public Stream<File> componentClassFiles() {
+                if (ComponentValidator.this.classes == null) {
+                    return Stream.empty();
+                }
+                return Stream.of(ComponentValidator.this.classes);
+            }
+        };
+
+        final Validators validators = Validators.build(configuration, helper);
+        final Set<String> errorsFromValidator = validators.validate(finder, components);
+        errors.addAll(errorsFromValidator);
 
         if (configuration.isValidateFamily()) {
             // todo: better fix is to get the package with @Components then check it has an icon
```

Comment:
it implies more changes (on BaseTask class), but; yes, it may be needed (lot of methods on BaseTask class are protected, but would better be public static on another class.)

Reply:
done, and also integrate ExtensionValidation